### PR TITLE
adding item builder to the text input

### PR DESCRIPTION
**Pull request by nimr77**

### CODE REVIEW

**Comment by nimr77** [comment]
I needed this since if my decoration has outlined border, in the dialog they will stick to each other, so I don't want my input box stick like this, 

in order to do so,  the itemBuilder method can be attached so I can add a padding to the input text

File: lib/src/text_input_dialog/ios_text_input_dialog.dart [hunk 1]
```diff
@@ -31,6 +33,7 @@ class IOSTextInputDialog extends StatefulWidget {
   final bool useRootNavigator;
   final WillPopCallback? onWillPop;
   final bool autoSubmit;
+  final Widget Function(Widget)? textInputBuilder;
```
Comment (mono0926) [inline lib/src/text_input_dialog/ios_text_input_dialog.dart:36]: `textFields` is List<DialogTextField>` and `textInputBuilder` isn't List, so I think it's not a good API.

File: lib/src/text_input_dialog/show_text_input_dialog.dart [hunk 2]
```diff
@@ -60,6 +62,7 @@ Future<List<String>?> showTextInputDialog({
         useRootNavigator: useRootNavigator,
         context: context,
         builder: (context) {
+          https: //github.com/nimr77/adaptive_dialog.git
```
Comment (mono0926) [inline lib/src/text_input_dialog/show_text_input_dialog.dart:65]: Should be deleted.

**Review by mono0926 — COMMENTED** [review]
While adaptive_dialog optimizes each UI on multiple platforms, it lacks some flexibility.
I do so intentionally, and recommend you to implement your own without adaptive_dialog if it doesn't fit your UI design.

File: lib/src/text_input_dialog/ios_text_input_dialog.dart (hunk 1 above)
Comment (nimr77) [reply lib/src/text_input_dialog/ios_text_input_dialog.dart:36]: Yes, but how I would change the text decoration, that's why I added the builder method

Comment (mono0926) [reply lib/src/text_input_dialog/ios_text_input_dialog.dart:36]: It certainly works, but I don't want to increase the complexity of this package by adding that customizability with its API design.

**Pull request closed without merging**